src/main/kotlin/com/sourcegraph/cody/edit/widget/LensWidgetGroup.kt

```diff
@@ -99,15 +110,26 @@ class LensWidgetGroup(val session: FixupSession, parentComponent: Editor) :
     }
   }
 
-  fun show(range: Range) {
-    commandCallbacks = session.commandCallbacks()
+  fun show(range: Range): CompletableFuture<Boolean> {
     val offset = range.start.toOffset(editor.document)
-    ApplicationManager.getApplication().invokeLater {
-      if (!isDisposed.get()) {
-        inlay = editor.inlayModel.addBlockElement(offset, false, true, 0, this)
-        Disposer.register(this, inlay!!)
+    val future = CompletableFuture<Boolean>()
+    onEventThread {
```

Comment:
That `val future = CompletableFuture<Boolean>()` seems unnecessary.
`onEventThread` already returns `CompletableFuture` which you could use, no?

Reply:
Ah good catch -- that was leftover from some refactoring.